include/alpaka/event/EventGenericSycl.hpp

```diff
@@ -81,7 +89,7 @@ namespace alpaka::traits
             experimental::QueueGenericSyclNonBlocking<TDev>& queue,
             experimental::EventGenericSycl<TDev>& event)
         {
-            event.m_event = queue.m_impl->get_last_event();
+            event.getNativeHandle() = queue.m_impl->get_last_event();
```

Comment:
This will not work. Maybe add a setter to `EventGenericSycl` for this case (or make the function a `friend`).

Reply:
Right, my bad. I think I would go for a setter, it seems safer to me.